x-pack/plugins/security_solution/public/common/components/page/manage_query.tsx

```diff
@@ -6,33 +6,53 @@
  */
 
 import type { Position } from '@elastic/charts';
-import { omit } from 'lodash/fp';
+import { noop, omit } from 'lodash/fp';
 import React, { useEffect } from 'react';
 
 import type { inputsModel } from '../../store';
 import type { GlobalTimeArgs } from '../../containers/use_global_time';
+import { useRefetchByRestartingSession } from './use_refetch_by_session';
+import { InputsModelId } from '../../store/inputs/constants';
+import { useIsExperimentalFeatureEnabled } from '../../hooks/use_experimental_features';
 
 export interface OwnProps extends Pick<GlobalTimeArgs, 'deleteQuery' | 'setQuery'> {
   headerChildren?: React.ReactNode;
   id: string;
+  inputId?: InputsModelId;
+  inspect?: inputsModel.InspectQuery;
   legendPosition?: Position;
   loading: boolean;
-  refetch: inputsModel.Refetch;
-  inspect?: inputsModel.InspectQuery;
+  refetch?: inputsModel.Refetch;
 }
 
 export function manageQuery<T>(
   WrappedComponent: React.ComponentClass<T> | React.ComponentType<T>
 ): React.FC<OwnProps & T> {
   const ManageQuery = (props: OwnProps & T) => {
-    const { loading, id, refetch, setQuery, deleteQuery, inspect = null } = props;
-    useQueryInspector({
-      queryId: id,
+    const isChartEmbeddablesEnabled = useIsExperimentalFeatureEnabled('chartEmbeddablesEnabled');
+
+    const {
+      deleteQuery,
+      id,
+      inputId = InputsModelId.global,
+      inspect = null,
       loading,
       refetch,
       setQuery,
+    } = props;
+    const { searchSessionId, refetchByRestartingSession } = useRefetchByRestartingSession({
+      inputId,
+      queryId: id,
+    });
+
+    useQueryInspector({
       deleteQuery,
       inspect,
+      loading,
+      queryId: id,
+      refetch: isChartEmbeddablesEnabled ? refetchByRestartingSession : refetch ?? noop, // refetchByRestartingSession is for refetching Lens Embeddables
```

Comment:
Why do we need to use this new `refetchByRestartingSession` function instead of the original `refetch`? Just to understand.
Also, `manageQuery` is used in many different places, some are not related to any lens visualizations, do we need to change the `refetch` to all those places?

Reply:
Lens Embeddables do not have a function we can call to refetch data like most of our components, it refetches when receiving a new search session ID. So in x-pack/plugins/security_solution/public/common/components/page/use_refetch_by_session.tsx Line 28 , we assign it the initial session ID, it's consumed in x-pack/plugins/security_solution/public/common/components/visualization_actions/lens_embeddable.tsx. It gets new session ID every time when refetching: x-pack/plugins/security_solution/public/common/components/page/use_refetch_by_session.tsx Line37

Yeah, and I think I should not have changed manage_query.
I put it in x-pack/plugins/security_solution/public/common/components/page/manage_query.tsx was trying not to repeat the implementation, but I think it makes more sense to be in the same place where each component gets their `refetch` function. Take KPI hosts as an example, I'm going to move `useRefetchByRestartingSession ` to `x-pack/plugins/security_solution/public/hosts/components/kpi_hosts/hosts/index.tsx`

```
const [loading, { refetch, id, inspect, ...data }] = useHostsKpiHosts({
    filterQuery,
    endDate: to,
    indexNames,
    startDate: from,
    skip: querySkip || isChartEmbeddablesEnabled,
  });

  const { searchSessionId, refetchByRestartingSession } = useRefetchByRestartingSession({
    inputId: InputsModelId.global,
    queryId: id,
  });

  return (
    <KpiBaseComponentManage
      data={data}
      id={id}
      inspect={inspect}
      loading={loading}
      fieldsMapping={fieldsMapping}
      from={from}
      to={to}
      updateDateRange={updateDateRange}
      refetch={isChartEmbeddablesEnabled ? refetchByRestartingSession : refetch}
      setQuery={setQuery}
      setQuerySkip={setQuerySkip}
      searchSessionId={isChartEmbeddablesEnabled ? searchSessionId : undefined}
    />
  );
```